Formula/azure-cli.rb

```diff
@@ -16,26 +13,493 @@ class AzureCli < Formula
+    ENV.prepend "LDFLAGS", "-L#{Formula["openssl"].opt_lib}"
+    ENV.prepend "CFLAGS", "-I#{Formula["openssl"].opt_include}"
+    ENV.prepend "CPPFLAGS", "-I#{Formula["openssl"].opt_include}"
+
+    # Get the CLI components we'll install
+    components = [
+      buildpath/"src/azure-cli",
+      buildpath/"src/azure-cli-core",
+      buildpath/"src/azure-cli-nspkg",
+      buildpath/"src/azure-cli-command_modules-nspkg",
+    ]
+    components += Pathname.glob(buildpath/"src/command_modules/azure-cli-*/")
 
-    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
-    bin.install_symlink Dir["#{libexec}/bin/*"]
-    (bash_completion/"azure").write Utils.popen_read("#{bin}/azure --completion")
+    # Install dependencies
+    # note: Even if in 'resources', don't include 'futures' as not needed for Python3
+    # and causes import errors. See https://github.com/agronholm/pythonfutures/issues/41
+    deps = resources.map(&:name).to_set - ["futures"]
+    deps.each do |r|
+      resource(r).stage do
+        system "python3", *Language::Python.setup_install_args(libexec)
+      end
+    end
+
+    # Install CLI
+    components.each do |item|
+      cd item do
+        system "python3", *Language::Python.setup_install_args(libexec)
+      end
+    end
+
+    # This replaces the `import pkg_resources` namespace imports from upstream
+    # with empty string as the import is slow and not needed in this environment.
+    File.open(site_packages/"azure/__init__.py", "w") {}
+    File.open(site_packages/"azure/cli/__init__.py", "w") {}
+    File.open(site_packages/"azure/cli/command_modules/__init__.py", "w") {}
+    File.open(site_packages/"azure/mgmt/__init__.py", "w") {}
+
+    az_exec = <<-EOS.undent
+      #!/usr/bin/env bash
+      export PYTHONPATH="#{ENV["PYTHONPATH"]}"
+      python#{xy} -m azure.cli \"$@\"
+    EOS
+    (bin/"az").write az_exec
+
+    bash_completion.install "az.completion" => "az"
+  end
 
-    # fix cxxstdlib warnings caused by installed (but not used) prebuild binaries for fibers
-    rm_rf Dir[libexec/"lib/node_modules/azure-cli/node_modules/fibers/bin/*-{46,48}"]
+  def caveats; <<-EOS.undent
+    This formula is for Azure CLI 2.0 - https://docs.microsoft.com/cli/azure/overview.
+    The previous Azure CLI has moved to azure-cli@1.0.
+    ----
+    Get started with:
+      $ az
+    EOS
   end
 
   test do
-    shell_output("#{bin}/azure telemetry --disable")
-    json_text = shell_output("#{bin}/azure account env show AzureCloud --json")
-    azure_cloud = JSON.parse(json_text)
-    assert_equal azure_cloud["name"], "AzureCloud"
-    assert_equal azure_cloud["managementEndpointUrl"], "https://management.core.windows.net"
-    assert_equal azure_cloud["resourceManagerEndpointUrl"], "https://management.azure.com/"
+    version_output = shell_output("#{bin}/az --version")
+    assert_match "azure-cli", version_output
+    system bin/"az", "account", "list"
```

Comment:
Can we assert the output matches something here? I'd like to just get rid of the version test above, and have this do something a bit more involved to make sure things are working properly. See the tests in `faas-cli`, `jose`, `jupyter`, and `stubby` for some good examples.

Reply:
I've added a different test.

Reply:
This is similar to https://github.com/Homebrew/homebrew-core/blob/master/Formula/azure-cli.rb#L35.